src/commands.cpp

```diff
@@ -8094,47 +7894,41 @@ CommandGetRegisteredContacts::CommandGetRegisteredContacts(MegaClient* client, c
     tag = client->reqtag;
 }
 
-void CommandGetRegisteredContacts::procresult()
+bool CommandGetRegisteredContacts::procresult(Result r)
 {
-    Error e;
-    if (checkError(e, client->json))
+    if (r.wasErrorOrOK())
     {
-        client->app->getregisteredcontacts_result(e, nullptr);
-        return;
+        client->app->getregisteredcontacts_result(r.errorOrOK(), nullptr);
+        return true;
     }
 
-    processResult(*client->app, client->json);
```

Comment:
Yes, I don't think processResult() was a very good approach, and it was only used in a few commands so far - I thought it was better to keep the same pattern as all the rest of the tests, especially while going through various revisions of parameter and return types for procresult().   I hope we can find a better way, that uses the commands in the way they were designed to be used (maybe by passing the request JSON in through the Reqest code) if we need unit tests for these - ideally the integration tests would cover these cases, in my opinion.

Reply:
Alright! :wink: 
I agree with you, let's keep your change and, in the future, prepare a different approach to exercise Unit Tests for commands.
As for covering all cases by integration tests, unfortunately, not everything is "integrate-able", since some commands require data from 2FA, from emails, etc. Of course, we can always spend significant amount of time to create a solution that also read emails or use a 2FA solution to generate PIN codes, but I think it's not worthy :)